packages/orchestration/src/utils/address.js

```diff
@@ -0,0 +1,46 @@
+// @ts-check
+import { Fail } from '@agoric/assert';
+
+/**
+ * @param {import('../types').ConnectionId} hostConnectionId
+ * @param {import('../types').ConnectionId} controllerConnectionId
+ * @param {object} [opts]
+ * @param {'ics27-1' | string} [opts.version] - default is `ics27-1`
+ * @param {'proto3' | string} [opts.encoding] - message encoding format for the channel. default is `proto3`
+ * @param {'ordered' | 'unordered'} [opts.ordering] - channel ordering. currently only `ordered` is supported for ics27-1
+ */
+export const makeICAConnectionAddress = (
+  hostConnectionId,
+  controllerConnectionId,
+  { version = 'ics27-1', encoding = 'proto3', ordering = 'ordered' } = {},
+) => {
+  hostConnectionId || Fail`hostConnectionId is required`;
+  controllerConnectionId || Fail`controllerConnectionId is required`;
+  const connString = JSON.stringify({
+    version,
+    controllerConnectionId, // self_connection_id
+    hostConnectionId, // counterparty_connection_id
+    address: '', // will be provided by the counterparty after channelOpenAck
+    encoding, //
+    txType: 'sdk_multi_msg',
+  });
+  return `/ibc-hop/${controllerConnectionId}/ibc-port/icahost/${ordering}/${connString}`;
```

Comment:
I'm somewhat surprised that JSON.stringify is the encoding. Why is `controllerConnectionId` included twice?

Are there any community tools for generating a URL like this? Or is this specific to Agoric?

a string template type could help document the provenance and context for this spec

Reply:
this suggests the https://github.com/Agoric/agoric-sdk/pull/9191/files#diff-20e9eca9ab68ad47bb9d4caa091036228bdd2ac215652dffebbdc1d27fbc80abR66 should be more flexible. I think this PR should land first tho